packages/data-store-json/src/data-store-json.ts

```diff
@@ -86,7 +119,15 @@ export class DataStoreJson implements IAgentPlugin {
       dataStoreDeleteVerifiableCredential: this.dataStoreDeleteVerifiableCredential.bind(this),
       dataStoreSaveVerifiablePresentation: this.dataStoreSaveVerifiablePresentation.bind(this),
       dataStoreGetVerifiablePresentation: this.dataStoreGetVerifiablePresentation.bind(this),
-      //dataStoreDeleteVerifiablePresentation: this.dataStoreDeleteVerifiablePresentation.bind(this),
+      dataStoreSaveMediationPolicy: this.dataStoreSaveMediationPolicy.bind(this),
+      dataStoreRemoveMediationPolicy: this.dataStoreRemoveMediationPolicy.bind(this),
+      dataStoreGetMediationPolicies: this.dataStoreGetMediationPolicies.bind(this),
+      dataStoreIsMediationGranted: this.dataStoreIsMediationGranted.bind(this),
+      dataStoreSaveMediation: this.dataStoreSaveMediation.bind(this),
+      dataStoreGetMediation: this.dataStoreGetMediation.bind(this),
+      dataStoreAddRecipientDid: this.dataStoreAddRecipientDid.bind(this),
+      dataStoreRemoveRecipientDid: this.dataStoreRemoveRecipientDid.bind(this),
+      dataStoreGetRecipientDids: this.dataStoreGetRecipientDids.bind(this),
```

Comment:
Thanks for this feedback. The dataStore changes definitely feel clunky in their current form and so I appreciate the guidance here and what you have outlined makes sense. 

With respect to implementing the `kv-store`, I couldn't see any current examples, so just wanted to outline my understanding here so it can be corrected early. 

**1. Implementation of the `kv-store`**
The `kv-store` should be configured in the agent.yml

**2. a new plugin should be created (e.g. `CoordinateMediationPolicy` or similar) **
The new plugin will require the `kv-store` and then be available on the agent (so shift all of the changes from the dataStore)

**3. Mediators operating on the V3 Coordinate Mediation specification install the plugin**
This means that the newly introduced v3 coordinate mediation message handler protocol implementation and the updated routing message handler protocol can access the `kv-store` with specific methods (such as the one you outlined above `coordinateMediationV3SavePolicy`). 

Thanks again for the guidance and I will check in on any feedback.

Reply:
Hi @mirceanis. Thanks again for the feedback. 

Just as an update, the following changes have been implemented in line with what has been discussed:

**1. `kv-store`**
The `kv-store` is now used for Mediation related storage with the use of the KeyValueTypeORMAdapter for persistent storage. This means that all previous changes to the `data-store` have been rolled back and are now encapsulated in the plugin (described below). 

**2. MediationManagerPlugin**
The PR now introduces a `MediationManagerPlugin` which allows us to expose methods pertaining to coordinate mediation on the agent. These methods are essentially ported like-for-like from what had been implemented on the `data-store` previously. 

**NOTE:**

A note on the `kv-store`. At present, there is no way to read out all of the keys in the `kv-store` for a given namespace. This is an issue because RECIPIENT_QUERY requires us to be able to list the `recipient_did`s. This constraint means that I have disabled the `cli` `mediate` `listPolicies`, and, more importantly, disabled the tests on coordinate medation v3 for RECIPIENT_QUERY. Do you have a proposal with respect to this? We could consider expanding the methods of the `kv-store` to allow for such a method to be exposed/adjust the existing `getMany` but first want to fully understand any considerations that you may have in mind. 

I look forward to your feedback. Thanks again.